fapolicy_analyzer/ui/subject_list.py

```diff
@@ -161,15 +162,14 @@ def __show_confirmation_dialog(self, subjects):
         confirm = confirmationDialog.run()
         confirmationDialog.destroy()
 
-        trust = subjects[0].trust.lower()
-
+        trust = self.find_db_trust(subjects[0])
         if confirm == 1:
             changeset = Changeset()
             paths = [subject.file for subject in subjects]
-            if trust == "u":
-                changeset.add_trust(*paths)
-            else:
+            if isinstance(trust, Trust):
                 changeset.del_trust(*paths)
+            else:
+                changeset.add_trust(*paths)
```

Comment:
I still don't think you logic is quite right here.  Not that you have the trust you need to look at the `trust.status` to determine the action.
* `trust` is None = Add
* `trust` is `t` and `subject.trust` is `st` = no action (there is nothing we can change about a file trusted in the system db)
* `trust` is `t` and `subject.trust` is `at` = Del
* `trust` is not `t` = Add

You can look at `TrustReconciliationDialog.__load_details` for how I'm setting the button visibility for this.